src/pocketmine/Player.php

```diff
@@ -811,6 +812,8 @@ protected function doFirstSpawn(){
 		$pk->z = $pos->z;
 		$this->dataPacket($pk);
 
+		$this->sendWeather($this->getLevel()->getWeatherManager()->getWeather());
```

Comment:
There are fields in StartGamePacket for this. If you can use those instead that would be better.